ngrinder-controller/src/main/java/org/ngrinder/agent/service/AgentManagerService.java

```diff
@@ -399,13 +402,17 @@ public AgentInfo getOne(Long id, boolean includeAgentIdentity) {
 	 */
 	@Transactional
 	public AgentInfo approve(Long id, boolean approve) {
-		AgentInfo found = agentManagerRepository.findOne(id);
-		if (found != null) {
-			found.setApproved(approve);
-			agentManagerRepository.save(found);
+		Optional<AgentInfo> found = agentManagerRepository.findOne(idEqual(id));
+		AgentInfo agentInfo = null;
+
+		if (found.isPresent()) {
+			agentInfo = found.get();
+			agentInfo.setApproved(approve);
+			agentManagerRepository.save(agentInfo);
```

Comment:
check if explicit ```save``` is necessary .

Reply:
hibernate auto saving for changed object.
so, this code is unnecessary.

Reply:
It's good. So you can reduce the code~